### PR TITLE
Add tari script transaction data structures

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
Added tari script transaction data structures as per RFC-0201 <https://rfc.tari.com/RFC-0201_TariScript.html>, used default tari script values everywhere:
- pub struct TransactionOutput
- pub struct TransactionInput
- pub struct Transaction
- pub struct BlockHeader
- All PRC interfaces
- All GRPC interfaces

Also updated comms wire mode to keep this network separate.

**TODO:**
- Update faucet to include new fields.
- Fix all unit tests.
- Migrate tari script into all transaction protocols.
- etc.

## Motivation and Context
See above.

## How Has This Been Tested?
Not tested, only compile without errors.

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [X] New feature (non-breaking change which adds functionality)
* [X] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by philipr-za — APPROVED** [review]
Looks fine as it is but there will certainly be many changes once we start actually using this work so I think this should go in and any issues we will fix in subsequent PRs

File: base_layer/core/src/blocks/block_header.rs [hunk 1]
```diff
@@ -106,6 +106,8 @@ pub struct BlockHeader {
     pub kernel_mmr_size: u64,
     /// Sum of kernel offsets for all kernels in this block.
     pub total_kernel_offset: BlindingFactor,
+    /// Sum of script offsets for all kernels in this block.
```
Comment (stringhandler) [inline base_layer/core/src/blocks/block_header.rs]:
```suggestion
    /// Sum of script offsets for all transactions in this block.
```

File: base_layer/core/src/blocks/new_blockheader_template.rs [hunk 2]
```diff
@@ -44,6 +44,8 @@ pub struct NewBlockHeaderTemplate {
     /// Total accumulated sum of kernel offsets since genesis block. We can derive the kernel offset sum for *this*
     /// block from the total kernel offset of the previous block header.
     pub total_kernel_offset: BlindingFactor,
+    /// Sum of script offsets for all kernels in this block.
```
Comment (stringhandler) [inline base_layer/core/src/blocks/new_blockheader_template.rs]:
```suggestion
    /// Sum of script offsets for all transactions in this block.
```

**Review by stringhandler — APPROVED** [review]
Looks good. There are a number of places where the comment for the script_offset is incorrectly mentioning a kernel, but they can be updated in a future pr

File: base_layer/core/src/blocks/new_blockheader_template.rs (hunk 2 above)
Comment (hansieodendaal) [reply base_layer/core/src/blocks/new_blockheader_template.rs]: Updated

File: base_layer/core/src/blocks/block_header.rs (hunk 1 above)
Comment (hansieodendaal) [reply base_layer/core/src/blocks/block_header.rs]: Updated

**Comment by stringhandler** [comment]
Looks like you still missed a few, but will merge

**Pull request merged**